controllers/gslb_controller_setup.go

```diff
@@ -0,0 +1,197 @@
+		Complete(r)
+}
+
+func (r *GslbReconciler) createGSLBFromIngress(c client.Client, a client.Object, strategy string) {
+	log.Info().
+		Str("annotation", fmt.Sprintf("(%s:%s)", strategyAnnotation, strategy)).
+		Str("ingress", a.GetName()).
+		Msg("Detected strategy annotation on ingress")
+
+	ingressToReuse := &netv1.Ingress{}
+	err := c.Get(context.Background(), client.ObjectKey{
+		Namespace: a.GetNamespace(),
+		Name:      a.GetName(),
+	}, ingressToReuse)
+	if err != nil {
+		log.Info().
+			Str("ingress", a.GetName()).
+			Msg("Ingress does not exist anymore. Skipping Glsb creation...")
+		return
+	}
+	gslbExist := &k8gbv1beta1.Gslb{}
+	err = c.Get(context.Background(), client.ObjectKey{
+		Namespace: a.GetNamespace(),
+		Name:      a.GetName(),
+	}, gslbExist)
+	if err == nil {
+		log.Info().
+			Str("gslb", gslbExist.Name).
+			Msg("Gslb already exists. Skipping Gslb creation...")
+		return
+	}
+	gslb := &k8gbv1beta1.Gslb{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:   a.GetNamespace(),
+			Name:        a.GetName(),
+			Annotations: a.GetAnnotations(),
+		},
+		Spec: k8gbv1beta1.GslbSpec{
+			Ingress: k8gbv1beta1.FromV1IngressSpec(ingressToReuse.Spec),
+		},
+	}
+
+	gslb.Spec.Strategy, err = r.parseStrategy(a.GetAnnotations(), strategy)
+	if err != nil {
+		log.Err(err).
+			Str("gslb", gslbExist.Name).
+			Msg("can't parse Gslb strategy")
+		return
+	}
+
+	err = controllerutil.SetControllerReference(ingressToReuse, gslb, r.Scheme)
+	if err != nil {
+		log.Err(err).
+			Str("ingress", ingressToReuse.Name).
+			Str("gslb", gslb.Name).
+			Msg("Cannot set the Ingress as the owner of the Gslb")
+	}
+
+	log.Info().
+		Str("gslb", gslb.Name).
+		Msg("Creating new Gslb out of Ingress annotation")
+	err = c.Create(context.Background(), gslb)
+	if err != nil {
+		log.Err(err).Msg("Glsb creation failed")
+	}
+}
+
+func (r *GslbReconciler) parseStrategy(annotations map[string]string, strategy string) (result k8gbv1beta1.Strategy, err error) {
+	toInt := func(k string, v string) (int, error) {
+		intValue, err := strconv.Atoi(v)
+		if err != nil {
+			return -1, fmt.Errorf("can't parse annotation value %s to int for key %s", v, k)
+		}
+		return intValue, nil
+	}
+
+	result = k8gbv1beta1.Strategy{
+		Type: strategy,
+	}
+
+	for annotationKey, annotationValue := range annotations {
+		switch annotationKey {
+		case dnsTTLSecondsAnnotation:
+			if result.DNSTtlSeconds, err = toInt(annotationKey, annotationValue); err != nil {
+				return result, err
+			}
+		case splitBrainThresholdSecondsAnnotation:
+			if result.SplitBrainThresholdSeconds, err = toInt(annotationKey, annotationValue); err != nil {
+				return result, err
+			}
+		case primaryGeoTagAnnotation:
+			result.PrimaryGeoTag = annotationValue
+		}
+	}
+
+	if strategy == depresolver.FailoverStrategy {
+		if len(result.PrimaryGeoTag) == 0 {
+			return result, fmt.Errorf("%s strategy requires annotation", depresolver.FailoverStrategy)
```

Comment:
This will print:
"failover strategy requires annotation" which is not very helpful imho

can we put here

"failover strategy requires annotation k8gb.io/primary-geotag to be also present on the Ingress resource"

(`primaryGeoTagAnnotation` == `k8gb.io/primary-geotag`)